src/skins/vector/views/organisms/CasLogin.js

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2015 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var React = require('react');
+
+var MatrixClientPeg = require('matrix-react-sdk/lib/MatrixClientPeg');
+
+module.exports = React.createClass({
+    displayName: 'CasLogin',
+
+    getInitialState: function() {
+        var splitLocation = window.location.href.split('/');
+        return {serviceUrl: splitLocation[0] + "//" + splitLocation[2]};
```

Comment:
Can you explain a bit more why you're doing this? `window.location.href` will be the entire URL including fragments, and we don't want to assume that the main URL path won't ever have `/`s in them (e.g. `vector.im/beta` is perfectly valid). Without any comments explaining your intent, I don't know if this is really what you want to be doing.